### PR TITLE
DFT Utilities [PR-1] [Config | Ztypes]

### PR DESCRIPTION
**Pull request by GreatRSingh**
## Description

Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/utils/dft_utils/config.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    Overload
+
+    Attributes
+    ----------
+    THRESHOLD_MEMORY: int
+        Threshold memory (matrix above this size should not be constructed)
+    CHUNK_MEMORY: int
+        The memory for splitting big tensors into chunks
+    VERBOSE: int
+        Verbosity level
```
Comment (shaipranesh2) [inline deepchem/utils/dft_utils/config.py:29]: Need more clarification on the attributes and how they help in context to the task you are gonna perform. (like verbosity level can be explained further)

File: deepchem/utils/dft_utils/test/test_config.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+"""
+Test for DFT Config Utilities
+"""
+
+from deepchem.utils.dft_utils.config import config
+
+
+def test_config():
+    conf = config
+    assert conf.THRESHOLD_MEMORY == 10 * 1024**3
```
Comment (shaipranesh2) [inline deepchem/utils/dft_utils/test/test_config.py:10]: are the config variables required to be constant? Or can the user try to change them as well?

**Comment by shaipranesh2** [comment]
@GreatRSingh I have some questions on this PR.
I understand you are trying to specify the config for the DFT tasks. But can the config variables become parameters for the DFT function that you will write? instead of reading it in a separate file like this?

File: deepchem/utils/dft_utils/test/test_config.py (hunk 2 above)
Comment (GreatRSingh) [reply deepchem/utils/dft_utils/test/test_config.py:10]: It looks like they used it as an overall configuration. I didn't find case of modification.

**Comment by GreatRSingh** [comment]
> @GreatRSingh I have some questions on this PR. I understand you are trying to specify the config for the DFT tasks. But can the config variables become parameters for the DFT function that you will write? instead of reading it in a separate file like this?

I think they had used it as an overall limit specifier for their DFT functions. so i guess one file should be better.

**Comment by GreatRSingh** [comment]
I think I understand your thought, we should have these configurations according to our own ML Hardware system.

File: deepchem/utils/dft_utils/test/test_config.py (hunk 2 above)
Comment (GreatRSingh) [reply deepchem/utils/dft_utils/test/test_config.py:10]: and I found some cases in which they have modified these.
`dqc/test/benchmarks.py: config.VERBOSE = 1`
and 2 more locations.

**Review by shaipranesh2 — APPROVED** [review]

File: deepchem/utils/dft_utils/test/test_config.py (hunk 2 above)
Comment (rbharath) [reply deepchem/utils/dft_utils/test/test_config.py:10]: Let's merge in for now, but we may need to come back later and clean this up

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**